### PR TITLE
QA Module - Tidying Code + Meeting BMRs

### PR DESCRIPTION
**Pull request by yoko-8**
The following changes were made:
- Did light refactoring of code to make it look cleaner
- Made small changes to meet BMRs: changing maximum char length of modal windows for nickname and email from 12 to 60, and having the answer list automatically refresh itself when an answer was successfully posted to the API.

### CODE REVIEW

File: client/src/components/qa/QAModule.jsx [hunk 1]
```diff
@@ -4,7 +4,6 @@ import SearchBar from './SearchBar';
 import QuestionList from './questionList/QuestionList';
 import AddQuestion from './addQuestion/AddQuestion';
 
-// eslint-disable-next-line react/prop-types
 export default function QAModule({ product }) {
   const [questionList, setQuestionList] = useState([]);
   const [sortList, setSort] = useState([]);
```
Comment (beanjermin) [inline client/src/components/qa/QAModule.jsx:9]: Alright, I just skimmed through your PR and it looks good to me tbh lol, but TOM (hello if you're reading this) wants to see comments so I'm going to try to nitpick a bit. sozz

File: client/src/components/qa/QAModule.jsx
Comment (beanjermin) [inline client/src/components/qa/QAModule.jsx:1]: In Line 57, I see that you're passing in setShow and setSort as props. 
From my understanding (and also bc a staff member previously told me at helpdesk), it's bad practice to pass down stateful functions as props. He told me if you need to pass down a fn, you should create a function and invoke the stateful fn in the parent component.

File: client/src/components/qa/SearchBar.jsx [hunk 2]
```diff
@@ -22,6 +22,6 @@ export default function SearchBar({ qList, setShowQs, qLen, setSort }) {
   }, [entry]);
 
   return (
-    <input className="searchBar" placeholder="Have a question? Search for answers..." onChange={(e) => setEntry(e.target.value)} />
+    <input className="searchBar" placeholder="Search your question..." onChange={(e) => setEntry(e.target.value)} />
```
Comment (beanjermin) [inline client/src/components/qa/SearchBar.jsx:25]: Just a suggestion, but I think 'Search Question' would suffice as a placeholder. Short, clean and simple.

File: client/src/components/qa/__tests__/qa.test.js [hunk 3]
```diff
@@ -90,13 +108,61 @@ describe('properly renders an answer', () => {
...
+  test('Add question button renders', () => {
+    render(<AddQuestion product={40349} />);
+    const button = screen.getByRole('button', { name: 'Ask Your Question' });
+    expect(button).toBeTruthy();
+  });
+  test('Add answer button renders', () => {
+    render(<AddAnswer id={328752} />);
+    const button = screen.getByRole('button', { name: 'Add Your Answer' });
+    expect(button).toBeTruthy();
+  });
```
Comment (beanjermin) [inline client/src/components/qa/__tests__/qa.test.js:167]: Wish I could give you more suggestions here, but I'm still a bit rusty on testing. Gonna look into it more tonight! Looks good to me though

File: client/src/components/qa/addAnswer/AddAnsModal.jsx [hunk 4]
```diff
@@ -2,7 +2,9 @@ import React, { useState } from 'react';
 import axios from 'axios';
 import Modal from '../../shared/Modal';
 
-export default function AddAnsModal({ show, setShow, id }) {
+export default function AddAnsModal({
+  show, setShow, id, getAllAnswers,
```
Comment (beanjermin) [inline client/src/components/qa/addAnswer/AddAnsModal.jsx]: This is the same thing I mentioned above, but just to reiterate: don't pass down stateful functions

File: client/src/components/qa/questionList/QuestionListEntry.jsx [hunk 5]
```diff
@@ -1,7 +1,6 @@
 import React, { useState } from 'react';
 import axios from 'axios';
 
-// eslint-disable-next-line react/prop-types
 export default function QuestionListEntry({ question }) {
   const [marked, setMarked] = useState(false);
   const [helpful, setHelpful] = useState(question.question_helpfulness);
```
Comment (beanjermin) [inline client/src/components/qa/questionList/QuestionListEntry.jsx:6]: kk I give up, honestly, this whole thing looked good to me LOl couldn't really find anything. Great work!

**Review by beanjermin — CHANGES_REQUESTED** [review]

File: client/src/components/qa/QAModule.jsx
Comment (yoko-8) [reply client/src/components/qa/QAModule.jsx:1]: Sounds good. I will make those changes and roll them out soon!

File: client/src/components/qa/questionList/AnswerList.jsx [hunk 6]
```diff
@@ -67,13 +65,13 @@ export default function AnswerList({ id }) {
                       <button type="submit">More Answers</button>
                     </form>
                   )
-                  : 'No More Answers'
+                  : ''
               }
-              <AddAnswer id={id} />
+              <AddAnswer id={id} getAllAnswers={getAllAnswers} />
```
Comment (beanjermin) [inline client/src/components/qa/questionList/AnswerList.jsx:70]: Great

**Review by beanjermin — APPROVED** [review]

**Pull request merged**